deps/wrapper.cc

```diff
@@ -162,6 +252,10 @@ JLCXX_MODULE define_julia_module(jlcxx::Module& mod)
     // attempting to use the shared library in Julia.
 
     mod.method("initialize_coreworker", &initialize_coreworker);
+    mod.method("initialize_coreworker_worker", [] (int node_manager, jlcxx::SafeCFunction julia_func) {
+        auto f = jlcxx::make_function_pointer<int()>(julia_func);
+        return initialize_coreworker_worker(node_manager, f);
+    });
```

Comment:
really struggling to follow what's happening here...we're creating a wrapped function named `initialize_coreworker_worker` that on the C++ side is a lambda, which takes two arguments: a node manager (as an int?) and a julia function (as a `SafeCFunction` which is a CxxWrap.jl thing).  then the lambda will create a pointer to the function that's suitable for passing to the `initialize_coreworker_worker` where it gets captured in the task execution callback and used to create the actual return value (regardless of what function descriptor is actually passed as part of the submitted task).

is that about right?

Reply:
Your understanding is correct! We could probably pass the `jlcxx::SafeCFunction` directly into the function but I also was looking into using C++ function pointers which work differently than how C works. Here is the CxxWrap documentation around [safe `cfunction`](https://github.com/JuliaInterop/CxxWrap.jl#safe-cfunction) for what I was working off of. Here's what I was reading on [C++ function pointers](https://www.cprogramming.com/tutorial/function-pointers.html).